packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationPersistenceService.java

```diff
@@ -77,25 +84,30 @@ public void save(ICause cause, IEditingContext editingContext, IRepresentation r
 
             String content = this.toString(representation);
 
-            var exists = this.representationDataSearchService.existsById(representationId);
+            var exists = this.representationMetadataSearchService.existsById(representationId);
 
             if (exists) {
                 var migrationData = this.getLastMigrationData(representation.getKind());
-                this.representationDataUpdateService.updateContentWithMigrationData(cause, representationId, content, migrationData.lastMigrationPerformed(), migrationData.migrationVersion());
+                this.representationContentUpdateService.updateContentByRepresentationIdWithMigrationData(cause, representationId, content, migrationData.lastMigrationPerformed(), migrationData.migrationVersion());
             } else {
                 var migrationData = this.getInitialMigrationData(representation.getKind());
-                var representationData = RepresentationData.newRepresentationData(representationId)
+                var representationMetadata = RepresentationMetadata.newRepresentationMetadata(representationId)
                         .project(AggregateReference.to(projectId))
                         .label(representation.getLabel())
                         .kind(representation.getKind())
                         .descriptionId(representation.getDescriptionId())
                         .targetObjectId(representation.getTargetObjectId())
+                        .build(cause);
+
+                var representationContent = RepresentationContent.newRepresentationContent(UUID.randomUUID())
```

Comment:
I understand using a random uuid here like some event sourcing system but it's a bit ambitious for now. Wouldn't that create a new snapshot in the database for each refresh? Couldn't you keep the representationId here in order to have only one content per metadata? We would instead perform an update instead of a new creation everytime it is necessary.

Reply:
If the representation already exists, we are using `RepresentationContentUpdateService` so I am not sure that it creates a new snapshot in the database for each refresh